eng/Packages.Data.props

```diff
@@ -86,16 +86,16 @@
     <PackageReference Update="Microsoft.Identity.Client.Extensions.Msal" Version="2.16.8" />
     <!-- TODO: Make sure this package is arch-board approved -->
     <PackageReference Update="System.IdentityModel.Tokens.Jwt" Version="5.4.0" />
+
+    <PackageReference Update="OpenTelemetry" Version="1.0.1"  Condition="'$(MSBuildProjectName)' == 'Azure.Monitor.OpenTelemetry.Exporter'" />
```

Comment:
Can you please add a comment saying something like `OpenTelemetry dependency approved for Azure.Monitor.OpenTelemetry.Exporter package only`

Reply:
I'm not sure this approach is any better then using VersionOverride. Is there a reason we went with this way?

Reply:
VersionOverrides are hard to see and review. This approach keeps the pattern of having centralized management but adds a limitation for some libraries.

Honestly, I wish we can disallow `VersionOverrides` for shipping packages altogether.

Reply:
OK. You could definitely add some checks for the VersionOverride if you wanted to add some validation around it.